tests/system.py

```diff
@@ -431,10 +431,12 @@ def test_delete_dataset_with_string(self):
         self.assertFalse(_dataset_exists(dataset_ref))
 
     def test_delete_dataset_delete_contents_true(self):
-        dataset_id = _make_dataset_id("delete_table_true")
+        dataset_id = _make_dataset_id("delete_table_true_with_content")
         project = Config.CLIENT.project
         dataset_ref = bigquery.DatasetReference(project, dataset_id)
-        dataset = retry_403(Config.CLIENT.create_dataset)(Dataset(dataset_ref))
+        dataset = retry_403(Config.CLIENT.create_dataset)(
+            Dataset(dataset_ref), exists_ok=True
```

Comment:
I'd rather not change the semantics of this call here:  if the dataset ID is unique (as it should be, and as your changes above imply), then we don't need `exists_ok=True`.